client/src/pokemon/PokemonInfo.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from "react";
+import { getPokemonByName } from "../helpers/pokemon-api";
+import styled from "@emotion/styled";
+
+const ULInline = styled.ul`
+  display: inline-block;
+`;
+
+class PokemonInfo extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: null
+    };
+
+    this.handleGettingData = this.handleGettingData.bind(this);
+    this.handleError = this.handleError.bind(this);
+  }
+
+  handleGettingData(data) {
+    this.setState({ data });
+  }
+
+  handleError(err) {
+    console.log("API Error:", err);
+  }
+
+  /* Life Cycle functions */
+  componentDidMount() {
+    const { match: params } = this.props;
+    if (params.pokemon) {
+      getPokemonByName(params.pokemon)
+        .then(this.handleGettingData)
+        .catch(this.handleError);
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const previousName = prevProps.match.params.pokemon;
+    const {
+      match: {
+        params: {
+          pokemon
+        }
+      }
+    } = this.props;
```

Comment:
next level destructuring lmao
Normally this is used for destructuring both sibling and nested props together. In this one case, a one-liner would've been fine.